src/Product/Image/Assignor.php

```diff
@@ -0,0 +1,96 @@
+<?php
```

Comment:
Surely that's doing something completely different, though. I dunno I don't think there's an issue with adding an extra class that does a single thing, rather than append an already bloated class.